src/main/java/org/stellar/sdk/ManageOfferOperation.java

```diff
@@ -125,10 +130,28 @@ public long getOfferId() {
      * @param price Price of 1 unit of selling in terms of buying.
      * @throws ArithmeticException when amount has more than 7 decimal places.
      */
-    public Builder(Asset selling, Asset buying, String amount, String price) {
+    public Builder(Asset selling, Asset buying, String amount, Price price) {
+      this.selling = checkNotNull(selling, "selling cannot be null");
+      this.buying = checkNotNull(buying, "buying cannot be null");
+      this.amount = checkNotNull(amount, "amount cannot be null");
+      this.price = checkNotNull(price, "price cannot be null");
+    }
+    
+    /**
+     * Creates a new ManageOffer builder. If you want to update existing offer use
+     * {@link org.stellar.sdk.ManageOfferOperation.Builder#setOfferId(long)}.
+     * @param selling The asset being sold in this operation
+     * @param buying The asset being bought in this operation
+     * @param amount Amount of selling being sold.
+     * @param price Price of 1 unit of selling in terms of buying.
+     * @throws ArithmeticException when amount has more than 7 decimal places.
+     */
+    @Deprecated
```

Comment:
I think we shouldn't deprecate this method. It's OK to use `String` price in many cases.